### PR TITLE
Switch back to the old psql-like datetime format

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Switch back to the old psql-like datetime format

## Implementation comments

I opted for a separate data class, to make this super explicit

## What is the intent behind these changes?

We reached a compromise with the NDMIS team that we would retain the
previous datetime format that psql gave us:

`2021-11-13 17:55:51.067867+00` in winter (GMT)
`2021-06-11 16:06:38.662521+00` in summer (BST)

The ISO format we use now is:

`2021-11-13T17:55:51.067867Z` in winter (GMT)
`2021-06-11T17:06:38.662521+01:00` in summer (BST)

This is because changing the ETL pipeline requires re-prioritisation of
other work, even for a datetime parsing change

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.ndmis.performance
+
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+// Agreement: the v2 NDMIS report uses this format, v3 switches to proper ISO8601
+// Deadline: stop producing v2 format after 13 December 2021
+data class NdmisDateTime(val dateTime: OffsetDateTime) {
+  override fun toString(): String {
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt:10]: This all hinges on the serialiser using `toString` -- is that the case?

Comment (tomsmyers) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt:10]: yes  - you can test this by adding to the `CsvLineAggregatorTest`

**Review by tomsmyers — APPROVED** [review]
nice solution

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.ndmis.performance
+
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+// Agreement: the v2 NDMIS report uses this format, v3 switches to proper ISO8601
+// Deadline: stop producing v2 format after 13 December 2021
```
Comment (tomsmyers) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt:8]: it didn't sound like the NDMIS team had signed off on this?

**Comment by tomsmyers** [comment]
one slight nit: i would probably have extended the OffsetDatetime class instead of wrapping it. this way it behaves the same in every other regard except for serialization to string.

**Comment by tomsmyers** [comment]
oh it's a final class.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt (hunk 2 above)
Comment (sldblog) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisDateTime.kt:8]: yeah, not yet. I'll amend the deadline when we agree

**Comment by sldblog** [comment]
thanks for the review + extra test!

**Pull request merged**